### PR TITLE
fix: Package flame_riverpod, setState() or markNeedsBuild() called during build.

### PR DESCRIPTION
**Pull request by markvideon**
# Description

The forceRebuild function now has an "isDirty" check to determine whether setState should be called. Previously, setState was uncritically invoked which *could* produce the Flutter framework error that occurs when setState is called while widgets are still being rebuilt. 

The setState call inside ref.watch has been replaced with a call to forceRebuild, and forceRebuild now has checks to prevent setState from being called while a build is in progress. A persistent frame callback has been added to reset supporting flags. 

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

N/A. Customer reported issue[ via Discord.](https://discord.com/channels/509714518008528896/1186421112217813062/1186908306179100723) 

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame_riverpod/lib/src/widget.dart [hunk 1]
```diff
@@ -47,14 +51,31 @@ class RiverpodAwareGameWidgetState<T extends Game> extends GameWidgetState<T>
   List<_ListenManual<Object?>>? _manualListeners;
 
   /// Rebuilds the [RiverpodAwareGameWidget] by calling [setState].
+  // Undesirable to call [setState] while the widget may be building.
+  // Honour requests to rebuild by setting a flag.
```
Comment (spydon) [inline packages/flame_riverpod/lib/src/widget.dart:55]: I guess this is meant as a comment and not a dartdoc? Never seen them mixed after each other before, does it even work? :sweat_smile:

**Review by spydon — APPROVED** [review]
Lgtm, just a comment about a comment

File: packages/flame_riverpod/lib/src/widget.dart (hunk 1 above)
Comment (spydon) [reply packages/flame_riverpod/lib/src/widget.dart:55]: Maybe the comments can be moved into the function?

File: packages/flame_riverpod/lib/src/widget.dart [hunk 2]
```diff
@@ -51,8 +51,9 @@ class RiverpodAwareGameWidgetState<T extends Game> extends GameWidgetState<T>
   List<_ListenManual<Object?>>? _manualListeners;
 
   /// Rebuilds the [RiverpodAwareGameWidget] by calling [setState].
-  // Undesirable to call [setState] while the widget may be building.
-  // Honour requests to rebuild by setting a flag.
+  /// As it is undesirable to call [setState] while the widget may be building,
+  /// this function honours (potential) subsequent requests to rebuild by
+  /// setting a flag.
```
Comment (spydon) [inline packages/flame_riverpod/lib/src/widget.dart:56]: Even better

**Review by spydon — APPROVED** [review]

File: packages/flame_riverpod/lib/src/widget.dart (hunk 1 above)
Comment (markvideon) [reply packages/flame_riverpod/lib/src/widget.dart]: I had considered that but have just modified the comments outside the function!

**Pull request merged**